src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -73,15 +73,17 @@ private static void EmitRegexPartialMethod(RegexMethod regexMethod, IndentedText
         private static void EmitRegexLimitedBoilerplate(
             IndentedTextWriter writer, RegexMethod rm, string reason)
         {
+            // If code generation is not supported but compilation is, then we fall back to use the compiled engine.
+            RegexOptions options = rm.Tree.Root.SupportsCompilation(out string? _) ? rm.Options | RegexOptions.Compiled : rm.Options; 
```

Comment:
I'm not sure if this is a good idea. It's how we initially had it, but I'd changed it because it means it takes the choice way from the developer whether they want the cached instance to incur the costs of compilation. If we don't do this, a dev can choose to set the Compiled option, but if we do this, they're forced to always use it.

Reply:
I see, sure I'm fine to remove it then. I only re-added it because I was under the impression that the reason why you removed it was because compiled engine and source generator engine had the same support restrictions, so if one wasn't supported the other one wouldn't either. Your comment about letting the dev decide is fine by me, and we can consider changing the diagnostic message we are currently emitting in case compiled engine is supported in order to suggest the user to use Compiled option in case they are using the pattern multiple times.

Reply:
> we can consider changing the diagnostic message we are currently emitting in case compiled engine is supported in order to suggest the user to use Compiled option in case they are using the pattern multiple times

Sounds reasonable. If you decide to do that, we'll probably want to exclude cases where it won't be used anyway, like NonBacktracking.

Reply:
I have reverted this change since there seems to be a problem in general about this diagnostic in particular since it doesn't end up displayed in Visual Studio. I logged issue https://github.com/dotnet/runtime/issues/68182 to investigate what is going on, and I will address the suggestion of RegexOptions.Compiled as part of that investigation outcome.